via/views/exceptions.py

```diff
@@ -61,16 +61,44 @@ def _get_meta(exception):
         if isinstance(exception, exception_type):
             return meta
 
-    return EXCEPTION_MAP.get(UnhandledException)
+    return EXCEPTION_MAP.get(UnhandledUpstreamException)
+
+
+@exception_view_config(
+    Exception,
+    route_name="proxy_google_drive_file",
+    renderer="via:templates/exception.html.jinja2",
+)
+@exception_view_config(
+    Exception,
+    route_name="proxy_google_drive_file:resource_key",
+    renderer="via:templates/exception.html.jinja2",
+)
+def google_drive_exceptions(exc, request):
+    """Catch all errors for Google Drive and display an HTML page."""
+
+    # Don't log 404's as they aren't very interesting
+    if not isinstance(exc, HTTPNotFound):
+        h_pyramid_sentry.report_exception()
```

Comment:
Can we continue logging 404s for now?

I think not logging those is slightly controversial as a 404 could mean "user tried to proxy a deleted file" or it could mean something else. Maybe we need to be a little bit more cautious and actually look in the response body and positively identify it as a "file does not exist". Even then we might be getting that error from Google due to a bug in our code (if something changed meaning our code for extracting IDs no longer works). Maybe make it a separate PR so we can discuss it separately.

Reply:
Also, does filtering out `HTTPNotFound` actually remove the 404s? Is that what gets raised?

Reply:
I was working on this separately: https://github.com/hypothesis/via/pull/617